dashboard/src/actions/repos.ts

```diff
@@ -80,10 +86,21 @@ export const fetchRepos = () => {
   };
 };
 
-export const installRepo = (name: string, url: string) => {
+export const installRepo = (name: string, url: string, authHeader: string) => {
   return async (dispatch: Dispatch<IStoreState>) => {
+    let auth;
+    if (authHeader.length) {
+      const secretName = `apprepo-${name}-secrets`;
+      await Secret.create(secretName, { authorizationHeader: btoa(authHeader) }, "kubeapps");
```

Comment:
based on the above recommendation on using ownerReferences, we _may_ need to create the Secret after the AppRepository, but it may also work asynchronously.